department-of-veterans-affairs/caseflow
# Jonathan/caseflow 4473 emo assign to rpo

Resolves https://vajira.max.gov/browse/CASEFLOW-4473

### Description
Implemented a new task action and subsequent modal for EMO users to successfully assign education predocket appeals to RPO

### Acceptance Criteria
- [ ] "Assign to Regional Processing Office" task action is created
- [ ] "Assign to RPO" modal is created
- [ ] Modal allows for assignment to any of the three available RPOs (Buffalo, Central Processing Office, Muskogee)
- [ ] Instructions are optional per designs
- [ ] EMO can successfully assign a predocket appeal to RPO
- [ ] Upon success, show the appropriate success banner (most likely a Success Alert in the Queue view however awaiting design feedback)
- [ ] Upon fail, show a failure (most likely a Slim Error Alert in the modal)
- [ ] After assignment the EMO task has a status of "on hold"
- [ ] The RPO task should be created and have a status of "assigned"

### Testing Plan
1. As a BVA Intake user, submit an Appeal with an Education issue and require pre-docketing.
2. Switch to an EMO user.
3. Navigate to the EMO organization queue: http://localhost:3000/organizations/edu-emo?tab=education_emo_unassigned&page=1
4. Ensure that the appeal you just created appears in the Unassigned tab.
5. Click on the appeal to go to the Case Details page.
6. Click on the Actions dropdown list and select "Assign to Regional Processing Office"
7. In the Modal, select a RPO and click the Submit button.
8. Verify that the success banner matches the RPO that you selected.
9. Click on the EMO Assigned Tab and verify that the appeal has moved over.
10. Switch users to an EMO RPO user.
11. Navigate to /queue and click the Switch Views button. Select the RPO that you assigned the appeal to in the previous modal.
12. Verify that the appeal appears in the RPO Assigned tab.
13. Open the rails console and inspect the Task Tree. Ensure that the appeal has the correct Tasks, Statuses and Assignees. 
![image](https://user-images.githubusercontent.com/98970951/165790070-78e57a57-6e8c-4e0e-9f83-a20dbd495210.png)




### User Facing Changes


 ![image](https://user-images.githubusercontent.com/98970951/165780399-26160114-b9ac-49eb-af06-e47947f2800c.png)
 ![local](https://user-images.githubusercontent.com/98970951/165779920-553ea94c-6103-48c0-9313-1db930ceedf4.png)
![image](https://user-images.githubusercontent.com/98970951/165786770-fc1b09e0-595d-4d67-bb8f-f15d9e56e8ea.png)



### Code Documentation Updates
